modules/post/multi/gather/enum_software_versions.rb

```diff
@@ -0,0 +1,139 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Multiplatform Installed Software Version',
+        'Description' => %q{
+          This module, when run against a compromised machine, will gather details on all installed software,
+          including their versions and if available, when they were installed, and will save it into a loot file for later use.
+          Users can then use this loot file to determine what additional vulnerabilites may affect the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'gwillcox-r7' ],
+        'Platform' => %w[win linux osx bsd solaris android],
+        # 'Platform'      => %w{ android osx win linux bsd solaris },
+        'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+    )
+  end
+
+  def store_linux_loot(listing)
+    file = store_loot('host.linux.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+    print_good("Stored information about the installed products to the loot file at #{file}")
+  end
+
+  # Run Method for when run command is issued
+  def run
+    case session.platform
+    when 'windows'
+      listing = cmd_exec('wmic product get Name, Description, Version, InstallDate', nil, 6000)
+      unless listing =~ /Description/
+        print_error('Was unable to get a listing of installed products...')
+        return nil
+      end
+      file = store_loot('host.windows.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+      print_good("Stored information about the installed products to the loot file at #{file}")
+    when 'linux'
+      # All of the following options were taken from https://distrowatch.com/dwres.php?resource=package-management
+      operating_system = cmd_exec('hostnamectl')
+      if operating_system =~ /not found/
+        print_error("Can't determine the host's OS, as the 'hostnamectl' command was not found")
+        return nil
+      elsif operating_system =~ /ubuntu/i || operating_system =~ /debian/i || operating_system =~ /elementary/i || operating_system =~ /mint/i || operating_system =~ /MX/ || operating_system =~ /zorin/i || operating_system =~ /kali/i
```

Comment:
Maybe factorize this into a non-grouping match, like `=~ (?:ubuntu|debian|elementary|…)/i`?

Reply:
Still learning Regex so not as familiar with the non-grouping match but your right this is an excellent example of when to use it. Thanks for the suggestion, will definitely implement this one 👍

Reply:
Thanks again @jvoisin for your suggestion, I'll be uploading a commit soon with these changes implemented :)